### DISCUSSION TITLE
谈 Kubernetes 的架构设计与实现原理 · /understanding-kubernetes

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#534] [Announcements]
https://draveness.me/understanding-kubernetes

**Comment by andyRon** [comment]
最近刚准备了解一下Kubernetes，  nice！

**Comment by JasperWey** [comment]
感谢

**Comment by langxuan** [comment]
控制器那一节第一句，“能够在集群的曾名”的“曾名”应为“层面”？

**Comment by draveness** [comment]
> @langxuan
> 控制器那一节第一句，“能够在集群的曾名”的“曾名”应为“层面”？

fixed

**Comment by cicicc** [comment]
您好，您的网站页面似乎出现了一些小的问题。首页中的分类，和点击old posts我昨晚和今天点击都会跳转至首页，内容也没有变化。
![imagepng](http://piv1n6e1j.bkt.clouddn.com//file/2018/11/2512e5e3cabc47d38c2e22d47ce98c4a_image.png)

**Comment by draveness** [comment]
> @cicicc
> 您好，您的网站页面似乎出现了一些小的问题。首页中的分类，和点击old posts我昨晚和今天点击都会跳转至首页，内容也没有变化。
> ![imagepng](http://piv1n6e1j.bkt.clouddn.com//file/2018/11/2512e5e3cabc47d38c2e22d47ce98c4a_image.png) 
> 
> 

不好意思，已经解决了

**Comment by ZD519** [comment]
求问笔者画图使用的工具是什么？看着很干净简约

**Comment by forezp** [comment]
您好，看了您的文章，收获颇多。这篇文章能否授权让我转载下，我的公众号目前有两万多粉丝，能够让更多的人读您的文章，非常感谢。

**Comment by Bitnut** [comment]
请教一下各位，目前有没有推荐的和 Kubernetes 配合使用的容器管理工具的项目啊？

**Comment by draveness** [comment]
> @Bitnut
> 请教一下各位，目前有没有推荐的和 Kubernetes 配合使用的容器管理工具的项目啊？

你说的是 Helm 这种包管理器么？

**Comment by Bitnut** [comment]
> > @Bitnut
> > 请教一下各位，目前有没有推荐的和 Kubernetes 配合使用的容器管理工具的项目啊？
> 
> 你说的是 Helm 这种包管理器么？

helm，也在用呢，不知道大佬常用的还有啥 :smile:

**Comment by yangzhihobg** [comment]
Kubernetes 遵循非常传统的客户端服务端架构？ 真的是这样吗？
主从架构难道和传统的客户端服务端架构是一个概念？

**Comment by draveness** [comment]
> Kubernetes 遵循非常传统的客户端服务端架构？ 真的是这样吗？
> 主从架构难道和传统的客户端服务端架构是一个概念？

了解一下 kubectl 和 apiserver 的工作原理吧

**Comment by lawrencewzen** [comment]
图画的好好看，讲解很清晰

**Comment by Formergg** [comment]
**架构**<br>
每一个 Kubernetes就集群都由一组 Master 节点和一系列的 Worker 节点组成，其中 Master 节点主要负责存储集群的状态并为 Kubernetes 对象分配和调度资源。

---
每一个K8s集群~就~都由一组xxxxxx（看起来感觉多了个"就"字）

---

*2020-07-08 UDPATES：已修复*

**Comment by XinruZhang** [comment]
条理清楚，信噪比高，深入浅出，感谢～

**Comment by cyenxchen** [comment]
**无侵入性** 主题那边的图挂了，博主可以补一下么？

---

*2020-09-19 UPDATES： 已修复*

**Comment by Huweicai** [comment]
大佬，给你提个意见，改改 hugo 模板博客加个侧边目录吧，可读性会好很多，一目了然~

**Comment by draveness** [comment]
> 大佬，给你提个意见，改改 hugo 模板博客加个侧边目录吧，可读性会好很多，一目了然~

不会写 CSS 啊

**Comment by dongxiem** [comment]
> @draveness
> > 大佬，给你提个意见，改改 hugo 模板博客加个侧边目录吧，可读性会好很多，一目了然~
> 
> 不会写 CSS 啊

大佬你好，你可以告诉大家有个Chrome插件叫 Smart Toc，可以显示出侧边栏。

**Comment by dashjay** [comment]
这几个概念太精辟了

- 声明式
- 显式接口
- 无侵入性
- 可移植性

**Comment by zhengyuhe01** [comment]
不能理解可移植性跟服务器是否有状态有什么关系

**Comment by draveness** [comment]
> @zhengyuhe01 不能理解可移植性跟服务器是否有状态有什么关系

有状态的服务在不同节点上迁移是很麻烦的，换个节点数据丢失是无法接受的，我们要考虑本地的数据怎么同步到其他节点上

**Comment by zhengyuhe01** [comment]
> @draveness
> > @zhengyuhe01 不能理解可移植性跟服务器是否有状态有什么关系
> 
> 有状态的服务在不同节点上迁移是很麻烦的，换个节点数据丢失是无法接受的，我们要考虑本地的数据怎么同步到其他节点上

难道k8s会没事的时候重启～～

**Comment by cyenxchen** [comment]
> @zhengyuhe01
> > @draveness
> > > @zhengyuhe01 不能理解可移植性跟服务器是否有状态有什么关系
> > 
> > 有状态的服务在不同节点上迁移是很麻烦的，换个节点数据丢失是无法接受的，我们要考虑本地的数据怎么同步到其他节点上
> 
> 难道k8s会没事的时候重启～～

不是说的k8s，是说的pod，你服务跑在pod上，运行服务的pod挂掉了，这期间没有持久化的数据就丢失了，而k8s此时会根据配置文件启动另外一个运行该服务的pod起来，但是因为服务挂掉丢失的数据也就丢了。

如果你的服务是无状态的，不需要持久化什么数据，那么挂了就挂了，在起一个或者多个就是了。

**Comment by zhengyuhe01** [comment]
> @CyshallChan
> > @zhengyuhe01
> > > @draveness
> > > > @zhengyuhe01 不能理解可移植性跟服务器是否有状态有什么关系
> > > 
> > > 有状态的服务在不同节点上迁移是很麻烦的，换个节点数据丢失是无法接受的，我们要考虑本地的数据怎么同步到其他节点上
> > 
> > 难道k8s会没事的时候重启～～
> 
> 不是说的k8s，是说的pod，你服务跑在pod上，运行服务的pod挂掉了，这期间没有持久化的数据就丢失了，而k8s此时会根据配置文件启动另外一个运行该服务的pod起来，但是因为服务挂掉丢失的数据也就丢了。
> 
> 如果你的服务是无状态的，不需要持久化什么数据，那么挂了就挂了，在起一个或者多个就是了。
> 
> 
pod为什么会挂掉：），无缘无故就挂了？

**Comment by cyenxchen** [comment]
> @zhengyuhe01
> > @CyshallChan
> > > @zhengyuhe01
> > > > @draveness
> > > > > @zhengyuhe01 不能理解可移植性跟服务器是否有状态有什么关系
> > > > 
> > > > 有状态的服务在不同节点上迁移是很麻烦的，换个节点数据丢失是无法接受的，我们要考虑本地的数据怎么同步到其他节点上
> > > 
> > > 难道k8s会没事的时候重启～～
> > 
> > 不是说的k8s，是说的pod，你服务跑在pod上，运行服务的pod挂掉了，这期间没有持久化的数据就丢失了，而k8s此时会根据配置文件启动另外一个运行该服务的pod起来，但是因为服务挂掉丢失的数据也就丢了。
> > 
> > 如果你的服务是无状态的，不需要持久化什么数据，那么挂了就挂了，在起一个或者多个就是了。
> > 
> > 
> pod为什么会挂掉：），无缘无故就挂了？
> 

pod挂了是因为你服务挂了，你是不是还要问服务为什么会挂？

**Comment by zhengyuhe01** [comment]
> @CyshallChan
> > @zhengyuhe01
> > > @CyshallChan
> > > > @zhengyuhe01
> > > > > @draveness
> > > > > > @zhengyuhe01 不能理解可移植性跟服务器是否有状态有什么关系
> > > > > 
> > > > > 有状态的服务在不同节点上迁移是很麻烦的，换个节点数据丢失是无法接受的，我们要考虑本地的数据怎么同步到其他节点上
> > > > 
> > > > 难道k8s会没事的时候重启～～
> > > 
> > > 不是说的k8s，是说的pod，你服务跑在pod上，运行服务的pod挂掉了，这期间没有持久化的数据就丢失了，而k8s此时会根据配置文件启动另外一个运行该服务的pod起来，但是因为服务挂掉丢失的数据也就丢了。
> > > 
> > > 如果你的服务是无状态的，不需要持久化什么数据，那么挂了就挂了，在起一个或者多个就是了。
> > > 
> > > 
> > pod为什么会挂掉：），无缘无故就挂了？
> > 
> 
> pod挂了是因为你服务挂了，你是不是还要问服务为什么会挂？
> 

厉害厉害，我自闭了

**Reply by jieqiyue** [reply]
你牛逼
